keymanager/src/crypto/kdf.rs

```diff
@@ -739,30 +935,35 @@ impl Kdf {
     /// concatenation of the key manager's runtime ID and the epoch, using ephemeral secret
     /// as the KMAC key.
     pub fn checksum_ephemeral_secret(
-        ephemeral_secret: &Secret,
         runtime_id: &Namespace,
+        secret: &Secret,
         epoch: EpochTime,
     ) -> Vec<u8> {
         let mut k = [0u8; 32];
 
         // KMAC256(ephemeral_secret, kmRuntimeID, epoch, 32, "ekiden-checksum-ephemeral-secret")
-        let mut f = KMac::new_kmac256(ephemeral_secret.as_ref(), &CHECKSUM_EPHEMERAL_SECRET_CUSTOM);
+        let mut f = KMac::new_kmac256(secret.as_ref(), &CHECKSUM_EPHEMERAL_SECRET_CUSTOM);
         f.update(runtime_id.as_ref());
         f.update(epoch.to_le_bytes().as_ref());
         f.finalize(&mut k);
 
         k.to_vec()
     }
 
-    fn new_d2() -> DeoxysII {
-        let mut seal_key = egetkey(Keypolicy::MRENCLAVE, MASTER_SECRET_SEAL_CONTEXT);
-        let d2 = DeoxysII::new(&seal_key);
-        seal_key.zeroize();
-
-        d2
+    fn derive_signing_key(runtime_id: &Namespace, secret: &Secret) -> signature::PrivateKey {
+        let sec = Inner::derive_secret(secret, &RUNTIME_SIGNING_KEY_CUSTOM, runtime_id.as_ref());
+        signature::PrivateKey::from_bytes(sec.0.to_vec())
     }
 }
 
+fn new_d2(key_policy: Keypolicy, context: &[u8]) -> DeoxysII {
```

Comment:
Should we just export the `new_d2` from `common::sgx::seal` as it seems identical? Or maybe with a slight rename to be more verbose what it is.